src/app/doctor/doctor.cpp

```diff
@@ -14,7 +14,7 @@ void Doctor::appendVisit(const std::shared_ptr<Visit>& visit)
 
 void Doctor::removeVisit(const std::shared_ptr<Visit>& visit)
 {
-    visits_.erase(std::remove(visits_.begin(), visits_.end(), visit));
+    visits_.erase(std::remove(visits_.begin(), visits_.end(), visit), visits_.end());
```

Comment:
Are we sure, that the range should be erased here?

Reply:
I was surprised, but it proved to work. Previous version had segfaults.